internal/service/cluster/api/server.go

```diff
@@ -0,0 +1,386 @@
+package api
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"net/url"
+	"strings"
+
+	api "github.com/openshift-kni/oran-o2ims/internal/service/cluster/api/generated"
+	"github.com/openshift-kni/oran-o2ims/internal/service/cluster/db/models"
+	"github.com/openshift-kni/oran-o2ims/internal/service/cluster/db/repo"
+	utils2 "github.com/openshift-kni/oran-o2ims/internal/service/cluster/utils"
+	"github.com/openshift-kni/oran-o2ims/internal/service/common/api/generated"
+	models2 "github.com/openshift-kni/oran-o2ims/internal/service/common/db/models"
+	"github.com/openshift-kni/oran-o2ims/internal/service/common/notifier"
+	"github.com/openshift-kni/oran-o2ims/internal/service/common/utils"
+)
+
+// ClusterServer implements StrictServerInterface. This ensures that we've conformed to the `StrictServerInterface` with a compile-time check
+var _ api.StrictServerInterface = (*ClusterServer)(nil)
+
+// ClusterServerConfig defines the configuration attributes for the resource server
+type ClusterServerConfig struct {
+	Address         string
+	CloudID         string
+	GlobalCloudID   string
+	BackendURL      string
+	Extensions      []string
+	ExternalAddress string
+}
+
+// ClusterServer defines the instance attributes for an instance of a cluster server
+type ClusterServer struct {
+	Config                   *ClusterServerConfig
+	Repo                     *repo.ClusterRepository
+	SubscriptionEventHandler notifier.SubscriptionEventHandler
+}
+
+// GetClusterResourceTypes receives the API request to this endpoint, executes the request, and responds appropriately
+func (r *ClusterServer) GetClusterResourceTypes(ctx context.Context, request api.GetClusterResourceTypesRequestObject) (api.GetClusterResourceTypesResponseObject, error) {
+	records, err := r.Repo.GetClusterResourceTypes(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get cluster resource types: %w", err)
+	}
+
+	objects := make([]api.ClusterResourceType, len(records))
```

Comment:
you need a 0 in between, otherwise, if you have 10 clusterresourcetype, you will end up with 20 objects. 10 empty ones

Reply:
i'm not doing `append` below though...

Reply:
sorry, I imagined the append. Please ignore